build_rust/src/main.rs

```diff
@@ -0,0 +1,264 @@
+            "qt:aqt:data:web:pages",
+        ])
+        .ensure_success()?;
+
+    create_dir_all(artifacts_dir.join("web"))?;
+
+    let web_dir = artifacts_dir.join("web");
+    for file in read_dir_files("anki/out/qt/_aqt/data/web/pages")? {
+        let file = file?;
+        let path = file.path();
+        copy_file(
+            &path,
+            web_dir.join(path.file_name().unwrap().to_str().unwrap()),
+        )?;
+    }
+    copy_file(
+        "anki/out/ts/reviewer/reviewer_extras_bundle.js",
+        artifacts_dir.join("web/reviewer_extras_bundle.js"),
+    )?;
+    copy_file(
+        "anki/cargo/licenses.json",
+        artifacts_dir.join("web/licenses-cargo.json"),
+    )?;
+    copy_file(
+        "anki/ts/licenses.json",
+        artifacts_dir.join("web/licenses-ts.json"),
+    )?;
+    Ok(())
+}
+
+fn build_android_jni() -> Result<()> {
+    println!("*** Building Android JNI library + backend interface");
+    let jni_dir = Path::new(ANDROID_OUT_DIR);
+    if jni_dir.exists() {
+        std::fs::remove_dir_all(jni_dir)?;
+    }
+    create_dir_all(jni_dir)?;
+
+    let all_archs = env::var("ALL_ARCHS").is_ok();
+    let ndk_targets = add_android_rust_targets(all_archs)?;
+    let (is_release, _release_dir) = check_release(false);
+
+    Command::run("cargo install cargo-ndk@3.2.0")?;
+
+    let mut command = Command::new("cargo");
+    command
+        // build products go into separate folder so they don't trigger recompile
+        // of robolectric/desktop code
+        .env("CARGO_TARGET_DIR", "target")
+        .env("STRINGS_JSON", env!("STRINGS_JSON_ANKIDROID"))
+        .arg("ndk")
+        .arg("-o")
+        .arg(jni_dir)
+        .args(ndk_targets)
+        .args(["build", "-p", "rsdroid"]);
+    if is_release {
+        command.arg("--release");
+    }
+    command.ensure_success()?;
+
+    Ok(())
+}
+
+// is_release, release/debug dir
+// windows robolectric is forced to release, as debug builds fail with an error
+fn check_release(force_release_on_windows: bool) -> (bool, &'static str) {
+    if env::var("RELEASE").is_ok() || (force_release_on_windows && cfg!(windows)) {
+        (true, "release")
+    } else {
+        (false, "debug")
+    }
+}
+
+/// Returns target list to pass to cargo ndk
+fn add_android_rust_targets(all_archs: bool) -> Result<&'static [&'static str]> {
+    Ok(if all_archs {
+        add_rust_targets(&[
+            "armv7-linux-androideabi",
+            "i686-linux-android",
+            "aarch64-linux-android",
+            "x86_64-linux-android",
+        ])?;
+        &[
+            "-t",
+            "armv7-linux-androideabi",
+            "-t",
+            "i686-linux-android",
+            "-t",
+            "aarch64-linux-android",
+            "-t",
+            "x86_64-linux-android",
+        ]
+    } else if cfg!(all(target_os = "macos", arch = "aarch64")) {
+        add_rust_targets(&["aarch64-linux-android"])?;
+        &["-t", "aarch64"]
```

Comment:
Fixes for macOS M1 build

```suggestion
    } else if cfg!(all(target_os = "macos", target_arch = "aarch64")) {
        add_rust_targets(&["aarch64-linux-android"])?;
        &["-t", "arm64-v8a"]
```